packages/sdk/src/oracle/types/state.ts

```diff
@@ -140,10 +140,20 @@ export type RequestIndex = RequestByEvent & { chainId: number };
 export type RequestIndexes = RequestIndex[];
 export type { RequestPrice, ProposePrice, DisputePrice, Settle };
 
+// combine all request data into a mega request object
+export type FullRequest = InputRequest &
+  // we cant assume any of these props will exist if we get event before query contract
+  Partial<Request> &
+  // take the more specific types from the Request type by omitting overlapping properties
+  Omit<
+    RequestIndex,
+    "proposedPrice" | "resolvedPrice" | "expirationTime" | "reward" | "finalFee" | "bond" | "customLiveness"
+  >;
+
```

Comment:
this new type is a mash of several different types, because we need to join data from several sources: 1. users requested input, 2. request state from contract and 3. known events for the request.  All of these will have slightly different properties.  Because this data can be set at any time, ie events can come in before contract query, or vice versa, we cant make strong assurances about the existence of any properties.